docs/posts/ibis-bench/index.qmd

```diff
@@ -0,0 +1,1103 @@
+con.raw_sql("PRAGMA disable_progress_bar;");
+```
+
+Now read the data and take a look at the first few rows:
+
+```{python}
+t = (
+    ibis.read_parquet(f"gs://{dir_name}/file_id=*.parquet")
+    .mutate(
+        timestamp=ibis._["timestamp"].cast("timestamp"),
+    )
+    .relocate(
+        "instance_type",
+        "system",
+        "sf",
+        "query_number",
+        "execution_seconds",
+        "timestamp",
+    )
+    .cache()
+)
+t.head()
+```
+
+We'll create a second table with details on each instance type including the CPU
+type, number of cores, and memory in gigabytes:
+
+```{python}
+#| code-fold: true
+#| code-summary: "Show code to get instance details"
+cpu_type_cases = (
+    ibis.case()
+    .when(
+        ibis._["instance_type"].startswith("n2d"),
+        "AMD EPYC",
+    )
+    .when(
+        ibis._["instance_type"].startswith("n2"),
+        "Intel Cascade and Ice Lake",
+    )
+    .when(
+        ibis._["instance_type"].startswith("c3"),
+        "Intel Sapphire Rapids",
+    )
+    .when(
+        ibis._["instance_type"] == "work laptop",
+        "Apple M1 Max",
+    )
+    .when(
+        ibis._["instance_type"] == "personal laptop",
+        "Apple M2 Max",
+    )
+    .else_("unknown")
+    .end()
+)
+cpu_num_cases = (
+    ibis.case()
+    .when(
+        ibis._["instance_type"].contains("-"),
+        ibis._["instance_type"].split("-")[-1].cast("int"),
+    )
+    .when(ibis._["instance_type"].contains("laptop"), 12)
+    .else_(0)
+    .end()
+)
+memory_gb_cases = (
+    ibis.case()
+    .when(
+        ibis._["instance_type"].contains("-"),
+        ibis._["instance_type"].split("-")[-1].cast("int") * 4,
+    )
+    .when(ibis._["instance_type"] == "work laptop", 32)
+    .when(ibis._["instance_type"] == "personal laptop", 96)
+    .else_(0)
+    .end()
+)
+
+instance_details = (
+    t.group_by("instance_type")
+    .agg()
+    .mutate(
+        cpu_type=cpu_type_cases, cpu_cores=cpu_num_cases, memory_gbs=memory_gb_cases
+    )
+).order_by("memory_gbs", "cpu_cores", "instance_type")
+
+cpu_types = sorted(
+    instance_details.distinct(on="cpu_type")["cpu_type"].to_pyarrow().to_pylist()
+)
+
+instance_details
+```
+
+### What's in the data?
+
+With the data, we can see we ran the benchmark on scale factors:
+
+```{python}
+sfs = sorted(t.distinct(on="sf")["sf"].to_pyarrow().to_pylist())
+sfs
+```
+
+:::{.callout-note title="What is a scale factor?" collapse="true"}
+A scale factor is roughly the size of the data in memory in gigabytes. For
+example, a scale factor of 1 means the data is roughly 1GB in memory.
+
+Stored on disk in Parquet format, the data is smaller -- about 0.38GB for scale
+factor 1 with the compression settings used in this benchmark.
+:::
+
+We can look at the total execution time by scale factor:
+
+```{python}
+#| code-fold: true
+#| code-summary: "Show bar plot code"
+c = px.bar(
+    t.group_by("sf").agg(total_seconds=t["execution_seconds"].sum()),
+    x="sf",
```

Comment:
I think this may look better if you convert the x-axis to a log base 2 scale.

Reply:
I didn't like it -- next time will add `sf=2` and `sf=4` and do this

Reply:
<img width="739" alt="image" src="https://github.com/ibis-project/ibis/assets/54814569/68bfcb79-ffea-45ae-8083-d700d7a83f98">